creusot/src/validate/terminates.rs

```diff
@@ -432,7 +438,11 @@ impl<'tcx> BuildFunctionsGraph<'tcx> {
 
         let impl_id = impl_id.to_def_id();
         let typing_env = ctx.typing_env(impl_id);
-        let term = ctx.term(item_id)?.unwrap();
+        let Some(term) = ctx.term(item_id)? else {
+            // ` ctx.term` can return None if the function we are searching for is not yet implemented.
+            // In this case, there is an Rust error happening: bail out and let the compiler report it normally.
+            return Err(CannotFetchThir);
```

Comment:
That's weird. Type checking is ran before this, isn't it? Then why are we here without all methods implemented?

Reply:
No, the termination check is ran in `before_analysis`, and at this stage type checking is lazy.

Reply:
That's unfortunate, because the code can be malformed in many ways which we may not anticipate, which would trigger other kinds of ICEs like this one.

Isn't there a way to make sure you're checking is fully completed? Is there another callback hook that we could use?

Reply:
We could try running in `after_analysis`, but I remember this causing issues as this needs to access thir? I'll try again just in case.